frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -47,35 +50,23 @@ interface PublicFormProviderProps {
   children: React.ReactNode
 }
 
-export const PublicFormProvider = ({
-  formId,
-  children,
-}: PublicFormProviderProps): JSX.Element => {
-  // Once form has been submitted, submission data will be set here.
-  const [submissionData, setSubmissionData] = useState<SubmissionData>()
+export function useCommonFormProvider(
+  formId: string,
+  data: PreviewFormViewDto | PublicFormViewDto | undefined,
+  error: ApiError | null,
+) {
   const [vfnTransaction, setVfnTransaction] =
     useState<FetchNewTransactionResponse>()
   const miniHeaderRef = useRef<HTMLDivElement>(null)
-  const { data, isLoading, error, ...rest } = usePublicFormView(
-    formId,
-    // Stop querying once submissionData is present.
-    /* enabled= */ !submissionData,
-  )
-  const { data: { captchaPublicKey } = {} } = useEnv(
-    /* enabled= */ !!data?.form.hasCaptcha,
-  )
-  const { hasLoaded, getCaptchaResponse, containerId } = useRecaptcha({
-    sitekey: data?.form.hasCaptcha ? captchaPublicKey : undefined,
-  })
-
-  const [cachedDto, setCachedDto] = useState<PublicFormViewDto>()
-
   const { createTransactionMutation } = useTransactionMutations(formId)
-  const { submitEmailModeFormMutation, submitStorageModeFormMutation } =
-    usePublicFormMutations(formId)
   const toast = useToast({ isClosable: true })
   const vfnToastIdRef = useRef<string | number>()
   const desyncToastIdRef = useRef<string | number>()
+  const isNotFormId = useMemo(() => !FORMID_REGEX.test(formId), [formId])
+
+  const [cachedDto, setCachedDto] = useState<
+    PreviewFormViewDto | PublicFormViewDto
+  >()
 
   useEffect(() => {
     if (data) {
```

Comment:
this should not be in the shared hook, otherwise there will be erroneous tracking of visits to public form even if it is a preview form. or we can add a param as to whether the current form is in preview mode...

Reply:
removed from shared hook!
Separately, I've also removed the other functions that depend on `cachedDto` because I didn't want to keep passing it back and forth.